### DISCUSSION TITLE
Decreasing MySQL wait_timeout value and connection pools

### DISCUSSION THREAD

**Discussion by idan3** [sqlalchemy/sqlalchemy#11657] [Usage Questions]
I want to decrease the wait_timeout variable value in my MySQL DB from the default (8h) to 30 minutes. I was wondering if it can be problematic if I use a connection pool, or the library knows how to handle it. If I get a timeout and the connection is closed does it know?

**Comment by CaselIT** [comment]
Hi,

You can use https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.pool_timeout to control this parameter

**Reply by idan3** [reply]
The thing is that I do not want to change anything in the code. I want to know if I change it from MySQL server itself if sqlalchemy will handle it correctly or it can cause a bug.

**Reply by CaselIT** [reply]
You may get errors about closed connections if you have no pool-timeout or poll-pre-ping, but you could have them also without changing that parameter since it seems it's 8h by default

**Reply by idan3** [reply]
But what would happen? It will return a closed connection? and then will keep returning it?

**Reply by CaselIT** [reply]
the best way is to try it.
The returned connection would be closed on the server side, so it would raise an error the first time you tried to issue a command on it. That would cause the connection to be invalidated, meaning that a new one would be returned the next time. But your application would have to handle that closed error

**Reply by idan3** [reply]
Now when I see the option you sent, "pool_timeout", I understand it is not what I need.
"number of seconds to wait before giving up on getting a connection from the pool".
It's not a connection timeout but how much time to wait for a connection from the pool, so it's not relevant anyway. 
Is there something to solve my problem?

**Reply by CaselIT** [reply]
sorry, I meant to link https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.pool_recycle

>  Is there something to solve my problem?

if your constraint is not to modify any code, it's unlikely that there is anything

**Reply by idan3** [reply]
Will it just drop the current connections and will Crete new ones when needed? (lazy)

**Reply by CaselIT** [reply]
if you mean pool recycle, yes, the old connection gets closed and a new one is created

**Reply by idan3** [reply]
Yes I understand. But lazy (only if needed) created? Or if pool has 15 connections than new 15 will be created imiddietly?

**Reply by CaselIT** [reply]
Only when needed

**Reply by idan3** [reply]
And Mysql closes them as well?

**Reply by idan3** [reply]
Does anyone know? I want to make sure these connections are not left 'hanging'?

**Reply by CaselIT** [reply]
What's the question?

**Reply by idan3** [reply]
I read again about your [suggestion](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.pool_recycle) and it doesn't seem like it's going to reduce the number of connections.
From the docs: "connection that has been open for more than one hour will be invalidated and replaced, upon next checkout".
If it happens only on checkout then it won't reduce the number of connections in the pool.

**Reply by CaselIT** [reply]
That should not be an issue, they are closed in the mysql server. What issue do they cause you those connections that are in the pool?
